receiver/awsecscontainermetricsreceiver/receiver.go

```diff
@@ -81,10 +85,23 @@ func (aecmr *awsEcsContainerMetricsReceiver) Shutdown(context.Context) error {
 }
 
 // collectDataFromEndpoint collects container stats from Amazon ECS Task Metadata Endpoint
-// TODO: Replace with acutal logic.
-func (aecmr *awsEcsContainerMetricsReceiver) collectDataFromEndpoint(ctx context.Context) error {
-	md := awsecscontainermetrics.GenerateDummyMetrics()
+func (aecmr *awsEcsContainerMetricsReceiver) collectDataFromEndpoint(ctx context.Context, typeStr string) error {
+	aecmr.provider = awsecscontainermetrics.NewStatsProvider(aecmr.restClient)
+	stats, metadata, err := aecmr.provider.GetStats()
 
-	err := aecmr.nextConsumer.ConsumeMetrics(ctx, internaldata.OCToMetrics(md))
-	return err
+	if err != nil {
+		aecmr.logger.Error("Failed to collect stats", zap.Error(err))
+		return err
+	}
+
+	mds := awsecscontainermetrics.MetricsData(stats, metadata, typeStr)
```

Comment:
Please add a TODO to add self metrics about the receiver that needs to be collected using `obsreport`.

Reply:
Updated.